internal/provider/resource_api_key.go

```diff
@@ -190,6 +195,7 @@ func refreshAPIKeyState(ctx context.Context, d *schema.ResourceData, m interface
 		"max_hits_per_query":          key.MaxHitsPerQuery,
 		"max_queries_per_ip_per_hour": key.MaxQueriesPerIPPerHour,
 		"referers":                    key.Referers,
+		"query_parameters":            key.QueryParameters,
```

Comment:
This is type mismatch because `QueryParameters` is not string.
We should convert to string with `transport.URLEncode(key.QueryParameters)`  like below.
https://github.com/algolia/algoliasearch-client-go/blob/40b333da0e6c734924e9ec40a831c63ad87f78bf/algolia/search/key.go#L42